src/containers/TaskRuns/TaskRuns.js

```diff
@@ -32,58 +31,82 @@ import {
   isRunning,
   labels,
   runMatchesStatusFilter,
-  urls
+  urls,
+  useWebSocketReconnected
 } from '@tektoncd/dashboard-utils';
 import { Add16 as Add, TrashCan32 as Delete } from '@carbon/icons-react';
 
 import { ListPageLayout } from '..';
 import { sortRunsByStartTime } from '../../utils';
-import { fetchTaskRuns } from '../../actions/taskRuns';
+import { fetchTaskRuns as fetchTaskRunsActionCreator } from '../../actions/taskRuns';
 
 import {
   getSelectedNamespace,
   getTaskRuns,
   getTaskRunsErrorMessage,
   isFetchingTaskRuns,
-  isReadOnly,
-  isWebSocketConnected
+  isWebSocketConnected,
+  isReadOnly as selectIsReadOnly
 } from '../../reducers';
 import { cancelTaskRun, deleteTaskRun, rerunTaskRun } from '../../api';
 
-const initialState = {
-  deleteError: null,
-  showDeleteModal: false,
-  toBeDeleted: []
-};
-
 const { CLUSTER_TASK, TASK } = labels;
 
-export /* istanbul ignore next */ class TaskRuns extends Component {
-  state = initialState;
+/* istanbul ignore next */
+function TaskRuns(props) {
+  const {
+    error,
+    fetchTaskRuns,
+    filters,
+    history,
+    intl,
+    isReadOnly,
+    kind,
+    loading,
+    namespace,
+    setStatusFilter,
+    statusFilter,
+    taskName,
+    taskRuns
+  } = props;
 
-  componentDidMount() {
+  const [deleteError, setDeleteError] = useState(null);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [toBeDeleted, setToBeDeleted] = useState([]);
+  const [cancelSelection, setCancelSelection] = useState(null);
+
+  useEffect(() => {
     document.title = getTitle({ page: 'TaskRuns' });
-    this.fetchTaskRuns();
-  }
+  }, []);
 
-  componentDidUpdate(prevProps) {
-    const { filters, namespace, webSocketConnected } = this.props;
-    const {
-      filters: prevFilters,
-      namespace: prevNamespace,
-      webSocketConnected: prevWebSocketConnected
-    } = prevProps;
-
-    if (namespace !== prevNamespace || !isEqual(filters, prevFilters)) {
-      this.reset();
-      this.fetchTaskRuns();
-    } else if (webSocketConnected && prevWebSocketConnected === false) {
-      this.fetchTaskRuns();
+  function fetchData() {
+    if (kind === 'ClusterTask') {
+      // TaskRuns from ClusterTask should have label 'tekton.dev/clusterTask=',
+      // (and that is the filter on the page), but some taskruns might still
+      // only have the old label 'tekton.dev/task='
+      // So, for ClusterTasks, also fetch with the old filter:
+      fetchTaskRuns({
+        filters: [`${TASK}=${taskName}`]
+      });
     }
+
+    fetchTaskRuns({ filters, namespace });
+  }
+
+  function reset() {
+    setDeleteError(null);
+    setShowDeleteModal(false);
+    setToBeDeleted([]);
   }
 
-  getError() {
-    const { error, intl } = this.props;
+  useEffect(() => {
+    reset();
+    fetchData();
+  }, [JSON.stringify(filters), namespace]);
+
+  useWebSocketReconnected(fetchData, namespace);
```

Comment:
`namespace` should be `webSocketConnected`, and you'll need to add it to the list of deconstructed props at the top of the component

Reply:
Ouch. Fixing it.

Reply:
@AlanGreene Fixed. :)